staging/src/k8s.io/kubectl/pkg/cmd/version/version.go

```diff
@@ -178,7 +178,7 @@ func (o *Options) Run() error {
 	return serverErr
 }
 
-func getKustomizeVersion() string {
+func GetKustomizeVersion() string {
```

Comment:
Given that the fallback constant at L184 is not tied to the correct value for other clients, should we export this, or should we factor out the Dep part and expose only that, e.g.:


```go
func getKustomizeVersion() string {
	if modVersion, ok := GetKustomizeModVersion(); ok {
		return modVersion 
	}
        return kustomizeVersion // other clients should provide their own fallback
}

func GetKustomizeModVersion() (string, bool) {
	info, ok := debug.ReadBuildInfo()
	if !ok {
		return "", false
	}
	for _, dep := range info.Deps {
		if dep.Path == "sigs.k8s.io/kustomize/kustomize/v4" {
			return dep.Version, true
		}
	}
	return "", false
}
```